packages/graphql/src/translate/translate-read-new.ts

```diff
@@ -0,0 +1,201 @@
+import createProjectionAndParams from "./create-projection-and-params";
+import { GraphQLWhereArg, GraphQLOptionsArg, GraphQLSortArg, Context, ConnectionField } from "../types";
+import createAuthAndParams from "./create-auth-and-params";
+import { AUTH_FORBIDDEN_ERROR } from "../constants";
+import createConnectionAndParams from "./connection/create-connection-and-params";
+
+function translateRead({ node, context }: { context: Context; node: Node }): [string, any] {
+    const { resolveTree } = context;
+    const { fieldsByTypeName } = resolveTree;
+    const whereInput = resolveTree.args.where as GraphQLWhereArg;
+    const optionsInput = resolveTree.args.options as GraphQLOptionsArg;
+
+    const hasOffset = Boolean(optionsInput?.offset) || optionsInput?.offset === 0;
+    const hasLimit = Boolean(optionsInput?.limit) || optionsInput?.limit === 0;
+
+    const labels = node.labelString;
+
+    const varName = "this";
+    const limitOrderVarName = "this_sort";
+    const matchStr = `MATCH (${varName}${labels})`;
+    let whereStr = "";
+    let authStr = "";
+    let offsetStr = "";
+    let limitStr = "";
+    let sortStr = "";
+    let projAuth = "";
+    let projStr = "";
+    let limitOrderProjStr = "";
+    let cypherParams: { [k: string]: any } = {};
+    const whereStrs: string[] = [];
+    const connectionStrs: string[] = [];
+
+    const projection = createProjectionAndParams({
+        node,
+        context,
+        fieldsByTypeName,
+        varName,
+    });
+    [projStr] = projection;
+    cypherParams = { ...cypherParams, ...projection[1] };
+    if (projection[2]?.authValidateStrs?.length) {
+        projAuth = `CALL apoc.util.validate(NOT(${projection[2].authValidateStrs.join(
+            " AND "
+        )}), "${AUTH_FORBIDDEN_ERROR}", [0])`;
+    }
+
+    if (projection[2]?.connectionFields?.length) {
+        projection[2].connectionFields.forEach((connectionResolveTree) => {
+            const connectionField = node.connectionFields.find(
+                (x) => x.fieldName === connectionResolveTree.name
+            ) as ConnectionField;
+            const connection = createConnectionAndParams({
+                resolveTree: connectionResolveTree,
+                field: connectionField,
+                context,
+                nodeVariable: varName,
+            });
+            connectionStrs.push(connection[0]);
+            cypherParams = { ...cypherParams, ...connection[1] };
+        });
+    }
+
+    if (whereInput) {
+        const where = createWhereAndParams({
+            whereInput,
+            varName,
+            node,
+            context,
+            recursing: true,
+        });
+        if (where[0]) {
+            whereStrs.push(where[0]);
+            cypherParams = { ...cypherParams, ...where[1] };
+        }
+    }
+
+    const whereAuth = createAuthAndParams({
+        operation: "READ",
+        entity: node,
+        context,
+        where: { varName, node },
```

Comment:
This might be an addition from another PR, or perhaps it was a property removed since I forked?

Reply:
This `translate-read-new.ts` file wasn't even meant to be committed, I was just using it as a scratch pad whilst trying to fit your changes into code, apologies! 🙈